include/zephyr/pm/policy.h

```diff
@@ -275,63 +267,140 @@ static inline int32_t pm_policy_next_event_ticks(void)
  * The system will not enter any power state that would make the system to
  * exceed the given latency value.
  *
+ * If immediate control manager is provided request may trigger an action asynchronous action
+ * to reconfigure the system runtime to fullfil requested latency. @p req contains an
+ * asynchronous notification field that must be initialize prior to the call to get the
+ * notification.
+ *
  * @param req Latency request.
  * @param value_us Maximum allowed latency in microseconds.
+ *
+ * @retval 0 if request is applied.
+ * @retval 1 if request required immediate action that is not completed. Configured completion
+ * notification will inform about completion.
+ * @retval -EIO immediate action returned error.
+ */
+int pm_policy_latency_request_add(struct pm_policy_latency_request *req, uint32_t value_us);
+
+/**
+ * @brief Synchronously add a new latency requirement.
+ *
+ * If immediate control manager is not used then it is an equivalent of
+ * @ref pm_policy_latency_request_add. If immediate control manager is used then context
+ * blocks until request is completed. It can only be called from the thread context.
+ * Asynchronous notification field in the @p req is configured and used internally.
+ *
+ * @retval 0 if request required immediate action that is not completed. Configured completion
+ * notification will inform about completion.
+ * @retval -EIO immediate action returned error.
  */
-void pm_policy_latency_request_add(struct pm_policy_latency_request *req,
-				   uint32_t value_us);
+int pm_policy_latency_request_add_sync(struct pm_policy_latency_request *req, uint32_t value_us);
```

Comment:
I would say the default function should be synchronous (without a suffix) and the special one, callable from IRQs, should be `_async()`.